src/components/MapInput/index.js

```diff
@@ -0,0 +1,120 @@
+import React, { useContext, useState, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import styled from '@datapunt/asc-core';
+import { Marker } from '@datapunt/react-maps';
+import { markerIcon } from 'shared/services/configuration/map-markers';
+import { location2feature } from 'shared/services/map-location';
+import MapContext from 'containers/MapContext/context';
+import { setLocationAction, setValuesAction } from 'containers/MapContext/actions';
+import Map from '../Map';
+import PDOKAutoSuggest from '../PDOKAutoSuggest';
+import reverseGeocoderService from './services/reverseGeocoderService';
+
+const Wrapper = styled.div`
+  position: relative;
+`;
+
+const StyledMap = styled(Map)`
+  height: 450px;
+  width: 100%;
+`;
+
+const StyledAutosuggest = styled(PDOKAutoSuggest)`
+  position: absolute;
+  top: 30px;
+  left: 20px;
+  width: calc(100% - 40px);
+  z-index: 401; // 400 is the minimum elevation were elements are shown above the map
+`;
+
+const MapInput = ({ className, value, onChange, mapOptions, ...otherProps }) => {
+  const { state, dispatch } = useContext(MapContext);
+  const [map, setMap] = useState();
+  const [marker, setMarker] = useState();
+  const { location, addressText: addressValue } = state;
+  const hasLocation = location && location?.lat !== 0 && location.lng !== 0;
+
+  useEffect(() => {
+    dispatch(setValuesAction(value));
+  }, [value, dispatch]);
+
+  const clickHandler = async e => {
+    dispatch(setLocationAction(e.latlng));
+    const response = await reverseGeocoderService(e.latlng);
+    dispatch(setValuesAction({
+      addressText: response.value,
+      address: response.data.address,
+    }));
+    onChange({ geometrie: location2feature(e.latlng), address: response.data.address });
+  };
+
+  const onSelect = useCallback(
+    option => {
+      const {
+        // eslint-disable-next-line no-shadow
```

Comment:
You're re-assigning a variable here; in this case I'd do
```jsx
const { data, value } = option;
dispatch(setValuesAction({ location: data.location, address: value }));
```

Reply:
no problem